velox/type/TimestampConversion.cpp

```diff
@@ -601,25 +592,31 @@ int64_t fromDateString(const char* str, size_t len) {
   return daysSinceEpoch;
 }
 
-int32_t castFromDateString(const char* str, size_t len, bool isIso8601) {
+int32_t castFromDateString(const char* str, size_t len, ParseMode mode) {
   int64_t daysSinceEpoch;
   size_t pos = 0;
 
-  auto mode =
-      isIso8601 ? ParseMode::kStandardCast : ParseMode::kNonStandardCast;
   if (!tryParseDateString(str, len, pos, daysSinceEpoch, mode)) {
-    if (isIso8601) {
+    if (mode == ParseMode::kStandardCast) {
       VELOX_USER_FAIL(
           "Unable to parse date value: \"{}\"."
           "Valid date string pattern is (YYYY-MM-DD), "
           "and can be prefixed with [+-]",
           std::string(str, len));
-    } else {
+    } else if (mode == ParseMode::kNonStandardCast) {
+      VELOX_USER_FAIL(
+          "Unable to parse date value: \"{}\"."
+          "Valid date string patterns include "
+          "(yyyy, yyyy-[m]m, yyyy-[m]m-[d]d, "
+          "yyyy-[m]m-[d]d *, yyyy-[m]m-[d]dT*), "
+          "and any pattern prefixed with [+-]",
+          std::string(str, len));
+    } else if (mode == ParseMode::kNonStandardNoTimeCast) {
```

Comment:
should we add an else with VELOX_UNREACHABLE() just in case?

Reply:
True, doesn't hurt. Added, and switched it to a switch case, seemed to make more sense here.